src/System.Runtime.Analyzers/Core/SystemRuntimeAnalyzersResources.resx

```diff
@@ -331,10 +331,10 @@
     <value>Specify StringComparison</value>
   </data>
   <data name="SpecifyStringComparisonDescription" xml:space="preserve">
-    <value>A string comparison operation uses a method overload that does not set a StringComparison parameter.</value>
+    <value>A string comparison operation uses a method overload that does not set a StringComparison parameter. If the result will be displayed to the user, such as when sorting a list of items for display in a list box, specify 'StringComparison.CurrentCulture' or 'StringComparison.CurrentCultureIgnoreCase' as the 'StringComparison' parameter. If comparing case-insensitive identifiers, such as file paths, environment variables, or registry keys and values, specify 'StringComparison.OrdinalIgnoreCase'. Otherwise, if comparing case-sensitive identifiers, specify 'StringComparison.Ordinal'.</value>
   </data>
   <data name="SpecifyStringComparisonMessage" xml:space="preserve">
-    <value>Specify StringComparison</value>
+    <value>Because the behavior of '{0}' could vary based on the current user's locale settings, replace this call in '{1}' with a call to '{2}'.</value>
```

Comment:
Minor grammar suggestion: avoid starting the sentence with "Because". You might want to just remove the term "because" in the diagnostic.